cmd/utils/git.go

```diff
@@ -94,6 +98,10 @@ func GetRandomSHA(ctx context.Context, path string) string {
 }
 
 func IsOktetoRepo() bool {
+
+	muOktetoSample.Lock()
```

Comment:
Why do we need this mutex?

Reply:
updated with comment

Reply:
I just want to point out that although setting a mutex "fixes" the race condition there is something fundamentally wrong about how this is implemented that needs to be revisited. Namely mutating global variables.

In the meantime, I think we can probably rewrite this with sync.Once and an atomic int. Playground: https://go.dev/play/p/iMeYELn_qyM

Reply:
thnks Fran, i've updated the code to use your feedback. I think i used once couple months ago but did not remember this when fixing the race.